cpp/src/parquet/encoding-benchmark.cc

```diff
@@ -163,4 +172,132 @@ static void BM_DictDecodingInt64_literals(benchmark::State& state) {
+    values_ = std::vector<ByteArray>();
+    values_.reserve(num_values_);
+    total_size_ = 0;
+
+    for (int64_t i = 0; i < binary_array.length(); i++) {
+      auto view = binary_array.GetView(i);
+      values_.emplace_back(static_cast<uint32_t>(view.length()),
+                           reinterpret_cast<const uint8_t*>(view.data()));
+      total_size_ += view.length();
+    }
+
+    valid_bits_ =
+        std::vector<uint8_t>(::arrow::BitUtil::BytesForBits(num_values_) + 1, 255);
+
+    auto encoder = MakeTypedEncoder<ByteArrayType>(Encoding::PLAIN);
+    encoder->Put(values_.data(), num_values_);
+    buffer_ = encoder->FlushValues();
+  }
+
+  void TearDown(const ::benchmark::State& state) override {}
+
+ protected:
+  int num_values_;
+  std::shared_ptr<::arrow::Array> input_array_;
+  uint64_t total_size_;
+  std::vector<ByteArray> values_;
+  std::vector<uint8_t> valid_bits_;
+  std::shared_ptr<Buffer> buffer_;
+};
+
+BENCHMARK_DEFINE_F(BM_PlainDecodingByteArray, DecodeArrow_Dense)
+(benchmark::State& state) {
+  while (state.KeepRunning()) {
+    auto decoder = MakeTypedDecoder<ByteArrayType>(Encoding::PLAIN);
+    decoder->SetData(num_values_, buffer_->data(), static_cast<int>(buffer_->size()));
+    ::arrow::internal::ChunkedBinaryBuilder builder(static_cast<int>(buffer_->size()),
+                                                    ::arrow::default_memory_pool());
+    decoder->DecodeArrow(num_values_, 0, valid_bits_.data(), 0, &builder);
+  }
+
+  state.SetBytesProcessed(state.iterations() * total_size_);
+}
+
+BENCHMARK_REGISTER_F(BM_PlainDecodingByteArray, DecodeArrow_Dense)->Range(1024, 65536);
+
+BENCHMARK_DEFINE_F(BM_PlainDecodingByteArray, DecodeArrowNonNull_Dense)
+(benchmark::State& state) {
+  while (state.KeepRunning()) {
+    auto decoder = MakeTypedDecoder<ByteArrayType>(Encoding::PLAIN);
+    decoder->SetData(num_values_, buffer_->data(), static_cast<int>(buffer_->size()));
+    ::arrow::internal::ChunkedBinaryBuilder builder(static_cast<int>(buffer_->size()),
+                                                    ::arrow::default_memory_pool());
+    decoder->DecodeArrowNonNull(num_values_, &builder);
+  }
+
+  state.SetBytesProcessed(state.iterations() * total_size_);
+}
+
+BENCHMARK_REGISTER_F(BM_PlainDecodingByteArray, DecodeArrowNonNull_Dense)
+    ->Range(1024, 65536);
+
+BENCHMARK_DEFINE_F(BM_PlainDecodingByteArray, DecodeArrow_Dict)
+(benchmark::State& state) {
+  while (state.KeepRunning()) {
+    auto decoder = MakeTypedDecoder<ByteArrayType>(Encoding::PLAIN);
+    decoder->SetData(num_values_, buffer_->data(), static_cast<int>(buffer_->size()));
+    ::arrow::BinaryDictionaryBuilder builder(::arrow::default_memory_pool());
+    decoder->DecodeArrow(num_values_, 0, valid_bits_.data(), 0, &builder);
+  }
+
+  state.SetBytesProcessed(state.iterations() * total_size_);
+}
+
+BENCHMARK_REGISTER_F(BM_PlainDecodingByteArray, DecodeArrow_Dict)->Range(1024, 65536);
```

Comment:
Consider utilizing named constants for the Range values to increase readability here. Same comment for all other similar invocations of Range in this diff.